src/main/xerus/monstercat/api/APIConnection.kt

```diff
@@ -77,20 +78,34 @@ class APIConnection(vararg path: String) : HTTPQuery<APIConnection>() {
 	/** Aborts this connection and thus terminates the InputStream if active */
 	fun abort() {
 		httpGet?.abort()
+		httpPost?.abort()
+		httpPut?.abort()
 	}
 	
 	// Direct Requesting
 	
 	private var httpGet: HttpGet? = null
-	fun execute() {
+	fun get(context: HttpClientContext? = null) {
 		httpGet = HttpGet(uri)
-		response = execute(httpGet!!)
+		response = execute(httpGet!!, context)
+	}
+	
+	private var httpPost: HttpPost? = null
```

Comment:
I don't think you need 3 separate variables for get, post and put, one should suffice since it's just for aborting the request

Reply:
b2a0cdfb6c79f566c4440650de445f2f5bd29baf